### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by glooms**
We could include a Makefile in our repository.

This is mainly to make testing a bit easier for developers but other things could be included as well.

Since most of our testing is through our integration test there might not be much of an incentive to track coverage since I don't think we can measure it since our tests are run with `os/exec`. We have very few unit tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+SHELL := /bin/bash
+
+start-deps:
+	docker-compose -f test/docker-compose.yml up -d
```
Comment (wennmo) [inline Makefile]: Since these are mainly used for development purposes maybe we should add `AcceptEULA=yes docker-compose...`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+SHELL := /bin/bash
+
```
Comment (wennmo) [inline Makefile]: Add a lint target as well and reuse it in CCI config?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+c.out:
+	./coverage.sh
+
+coverage: c.out
+	go tool cover -html=c.out -o coverage.html
+	rm c.out
+
+example:
+	@echo "Starting engine in docker"
+	docker-compose -f examples/docker-compose.yml up -d
```
Comment (wennmo) [inline Makefile]: Same here, this will only work if you have the env `AcceptEULA` set.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	@./corectl generate-docs
+	@./corectl generate-spec
+
+lint:
+	go fmt ./...
+	@$(shell which golint || go get -u golang.org/x/lint/golint)
+	golint -set_exit_status
+
+start-deps:
+	AcceptEUAL=yes docker-compose -f test/docker-compose.yml up -d
```
Comment (wennmo) [inline Makefile]: typo :)

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+c.out:
+	./coverage.sh
+
+coverage: c.out
+	go tool cover -html=c.out -o coverage.html
+	rm c.out
+
+example:
+	@echo "Starting engine in docker"
+	AcceptEUAL=yes docker-compose -f examples/docker-compose.yml up -d
```
Comment (wennmo) [inline Makefile]: typo

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	@echo "Building corectl"
+	go build
+	./corectl build --config examples/corectl.yml
+
+.PHONY: test docs
```
Comment (wennmo) [inline Makefile]: Do we need these phonys?

File: main.go [hunk 7]
```diff
@@ -6,7 +6,9 @@ import (
 
 // version will be set with: go build -ldflags "-X main.version=X.Y.Z"
 var version = "development build"
```
Comment (wennmo) [inline main.go]: I guess we can leave `version` empty now as well

File: Makefile (hunk 6 above)
Comment (glooms) [reply Makefile]: Yes, since we have a `test` and `docs` folder we need phony so that make runs the build target regardless of whether they exist.

**Pull request merged**